pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -186,16 +186,17 @@ func (r *ReconcileNodeMaintenance) handleLease(specAnnoData *ReconcileNodeMainte
 	switch leaseStatus {
 	case LeaseStatusExpired:
 		{
-			if err := r.client.Delete(context.TODO(), lease); err != nil {
-				r.reqLogger.Infof("failed to delete expired lease. node: %s err: %v\n", nodeName, err)
-
+			if err := r.createOrUpdateLease(node, specAnnoData.RequestedMaintenanceWindowSeconds, lease); err != nil {
```

Comment:
``lease`` is always ``nil`` here.
Seems like this case and ``LeaseStatusNotFound`` could be handled together.

Reply:
there is a different in updating an existing lease (lease not nil, called when LeaseStatusExpired) and creating a new one (lease is nil ; in caller LeaseStatusNotFound); isn't there?  this is what the lease ref is indicating. How can they be handled together?

Reply:
``r.doesLeaseExist`` was returning `lease=nil` in both cases